### PR TITLE
Arnej/detect hostname and more

### PR DESCRIPTION
**Pull request by arnej27959**
@mpolden please review
not tested on windows; also needs testing in IPv6-only environment.
ideally various broken network setups should be checked also.

@havardpe could maybe take a look and compare with C++ code for hostname detection and validation.

Adds:
```
func ExportDefaultEnvToSh() error
func FindOurHostname() (string, error)
func FindVespaUser() string
func HasOnlyIpV6() bool
```

### CODE REVIEW

**Comment by havardpe** [comment]
C++ does canonical hostname lookup based on hostname while the go code seems to do reverse lookup based on ip address instead. The C++ code requires that all ip addresses returned by the dns lookup of a potential hostname actually belongs to us; the go code seems a bit more relaxed on this one. There does not seem to be any special handling of loopback interfaces in C++

File: client/go/vespa/detect_hostname_test.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDetectHostname(t *testing.T) {
+	os.Setenv("VESPA_HOSTNAME", "foo.bar")
```
Comment (mpolden) [inline client/go/vespa/detect_hostname_test.go:14]: Consider using `t.Setenv`.

File: client/go/vespa/load_env_test.go [hunk 2]
```diff
@@ -98,3 +99,59 @@ override VESPA_V2 v2
...
+	setup(t, `
+override VESPA_USER unprivuser
+`)
+	LoadDefaultEnv()
+	u = FindVespaUser()
+	assert.Equal(t, "unprivuser", u)
+}
+
+func TestExportEnv(t *testing.T) {
+	os.Setenv("VESPA_FOO", "was foo")
```
Comment (mpolden) [inline client/go/vespa/load_env_test.go:120]: Consider using `t.Setenv` which automatically clears the variables after this test finishes.

**Review by mpolden — APPROVED** [review]

**Review by mpolden — APPROVED** [review]
👍

**Pull request merged**